landing_page/spec/controllers/user_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+    context "create a new user account with acceptable credentials" do
+
+        it "successfully adds a new user" do
+            expect{create_new_user}.to change(User, :count).by(1)
+        end
+
+        it "logs the user in upon account creation" do
+            create_new_user
+            expect(session[:user_id]).to_not be_nil
+        end
+
+        it "redirects a new user to their profile page upon creation" do
+            create_new_user
+            expect(response).to redirect_to(assigns(:user))
+        end
+
+    end
+
+    context "GET #index" do
+
+        it "shows the list of users, expects a redirect response to be 200" do
+            get :index
+            expect(response.status).to eq(200)
+        end
+
+    end
+
+    context "GET #show" do
+
+        it "shows a user their profile, expect redirect response to be 200" do
+            user = FactoryBot.create :user
+            user.save
+            login(user)
+
+            get :show, params: { id: user.id }
+            expect(response.status).to eq(200)
+        end
+
+    end
+
+    context "DELETE #destroy" do
+
+        it "deletes a user from the database" do
+            user = FactoryBot.create :user
+            user.save
+            login(user)
+
+            expect{delete :destroy, params: { id: user.id }}.to change(User, :count).by(-1)
+        end
+
+    end
+
+    context "PUT #update" do
+
+        it "allows a user to update their data" do
+            user = FactoryBot.create :user
```

Comment:
Although the expectation is correct, checking the name that was provided in the `:update` call, it's a good practice to make the setup more explicit. Something along the lines of:

```suggestion
            user = FactoryBot.create :user, name: "Not Jane"
```

That way it's clear where the reader should focus on.

Reply:
Resolved in commit commit d97be7bed27ec18d0421cdb237a47d222374017a